source/speech/speech.py

```diff
@@ -78,12 +79,31 @@
 _curWordChars: List[str] = []
 
 
-class SpeechMode(IntEnum):
+class SpeechMode(DisplayStringIntEnum):
 	off = 0
 	beeps = 1
 	talk = 2
 	onDemand = 3
 
+	@property
+	def _displayStringLabels(self) -> dict[Self, str]:
+		return {
+			# Translators: Name of the speech mode which disables speech output.
+			self.off: pgettext("speechModes", "off"),
+			# Translators: Name of the speech mode which will cause NVDA to beep instead of speaking.
+			self.beeps: pgettext("speechModes", "beeps"),
+			# Translators: Name of the speech mode, which, when enabled, causes NVDA to talk as normal.
+			self.talk: pgettext("speechModes", "talk"),
+			# Translators: Name of the on-demand speech mode;

```

Comment:
Speech modes were described in the comments for translators previously, so I see no reason to make these comments less detailed. I've applied your suggestion, though couldn't attribute it to you since this line had to be reformatted, to keep with in line length limit.